prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,298 @@
+                    i, newdata.columns.get_loc("trip start battery charge")
+                ] = kwh
+                trip_num = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip number")] = trip_num
+
+            if (
+                newdata.iloc[i, newdata.columns.get_loc("total vehicle miles traveled")]
+                < veh_range * const.ER
+            ):
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip end battery charge")] = (
+                    newdata.iloc[
+                        i, newdata.columns.get_loc("trip start battery charge")
+                    ]
+                    - newdata.iloc[i, newdata.columns.get_loc("Miles traveled")]
+                    * kwhmi
+                    * const.ER
+                )
+            else:
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 0
+
+            trip_data = newdata.iloc[i]
+
+            newdata.iloc[
+                i, newdata.columns.get_loc("charging power")
+            ] = charging.get_charging_power(
+                power,
+                trip_strategy,
+                location_strategy,
+                kwh,
+                trip_data,
+            )
+
+            newdata.iloc[
+                i, newdata.columns.get_loc("charging time")
+            ] = charging.get_charging_time(
+                newdata.iloc[i, newdata.columns.get_loc("charging power")],
+                kwh,
+                newdata.iloc[i, newdata.columns.get_loc("trip end battery charge")],
+                const.charging_efficiency,
+                trip_data,
+            )
+
+            newdata.iloc[i, newdata.columns.get_loc("charging consumption")] = (
+                newdata.iloc[i, newdata.columns.get_loc("charging power")]
+                * newdata.iloc[i, newdata.columns.get_loc("charging time")]
+                * const.charging_efficiency
+            )
+
+            # charging start point
+            start_point = round(
+                newdata.iloc[i, newdata.columns.get_loc("End time (hour decimal)")]
+                * 100
+            )
+
+            # charging end point
+            end_point = start_point + round(
+                newdata.iloc[i, newdata.columns.get_loc("charging time")] * 100
+            )
+            weekday_electricload[start_point:end_point] += newdata.iloc[
+                i, newdata.columns.get_loc("charging power")
+            ]
+
+    weekday_output_electricload = np.zeros(48)
+    for k in range(48):
+        if k == 0:
+            weekday_output_electricload[k] = sum(weekday_electricload[:100]) / 100
+        elif k == 47:
+            weekday_output_electricload[k] = sum(weekday_electricload[4700:]) / 100
+        else:
+            weekday_output_electricload[k] = (
+                sum(weekday_electricload[(k + 1) * 100 - 150 : (k + 1) * 100 - 50])
+                / 100
+            )
```

Comment:
I think this block can also be defined as a function that takes in a length-4800 array and down-samples to length 48, so that it doesn't need to get repeated for both weekdays and weekends.